### ISSUE TITLE
Entities don't seem to be available during test

### ISSUE THREAD

**Issue by dustinleblanc** [guncha25/drupal-codeception#1]
Hey @guncha25!

Thanks for taking the time to create this codeception module!

I have just started using to test an decoupled drupal 8 site using Contenta and I seem to be having some issues getting the functionality to work. I am trying to setup test entities like this in my test:

```php
$I->createEntity([
        'title' => 'Cold Cash, Colder Hearts',
        'type' => 'poems',
        'body' => 'They are sick, they are poor\nAnd they die by the thousands and we look away\nThey are wolves at the door\nAnd they\'re not gonna move us or get in our way',
        'field_poem_a_day_date' => time()
      ], 'node');
```

and then I am making assertions that the node itself is in the output:

```php
      $I->sendGET('/poem-a-day');
      $I->seeResponseCodeIs(200);
      $I->seeResponseContains('They are sick');
```

The output looks like this when I run it:

```
1) RetrievePoemADayCest: Retrieve poem a day via api
 Test  tests/api/RetrievePoemADayCest.php:retrievePoemADayViaApi
 Step  See response contains "They are sick"
 Fail  REST response contains
Failed asserting that '{REALLY LONG RESPONSE CONTAINING OTHER ITEMS BUT NOT MY ITEM}' contains "They are sick".

Scenario Steps:

 4. $I->seeResponseContains("They are sick") at tests/api/RetrievePoemADayCest.php:17
 3. $I->seeResponseCodeIs(200) at tests/api/RetrievePoemADayCest.php:16
 2. $I->sendGET("/poem-a-day") at tests/api/RetrievePoemADayCest.php:15
 1. $I->createEntity({"title":"Cold Cash, Colder Hearts","type":"poems","body":"They are sick, they are poor\\nAnd they die by the thousands and we look away\\nThey are wolves at the door\\nAnd they're not gonna move us or get in our way","fiel...},"node") at tests/api/RetrievePoemADayCest.php:14


FAILURES!
Tests: 2, Assertions: 6, Failures: 1.
Script codecept run handling the test event returned with error code 1
```

This could very well be that my stub node is just created incorrectly, I can't seem to find my records in the database even if I tell the module not to clean up after the test. This leads me to believe that the createEntity step is actually failing because I am passing it bad field data of some kind but just not reporting the failure. I fully expect I am doing something wrong, I am just not sure what it is.

**Comment by guncha25** [comment]
@dustinleblanc createEntity method returns entity itself if it was created successfully. In this method test will fail if any exception is thrown, but in case there where no exception but entity still was not created bool FALSE will be returned. So you can validate that by: 
```
$node = $I->createEntity([
        'title' => 'Cold Cash, Colder Hearts',
        'type' => 'poems',
        'body' => 'They are sick, they are poor\nAnd they die by the thousands and we look away\nThey are wolves at the door\nAnd they\'re not gonna move us or get in our way',
        'field_poem_a_day_date' => time()
      ]);
$I->assertNotFalse($node, 'Node should be created.')
```
FYI - Node is default entity so you don't have to specify that.

**Comment by dustinleblanc** [comment]
@guncha25 thanks for the tip! I'll try checking that. I'll go ahead and close this issue. Thanks again for creating this module!

**Comment by dustinleblanc** [comment]
@guncha25 I figured out the issue, I had set some field values that weren't valid. I think we might be able to catch this in future calls by calling ->validate() on the entity before saving it?

**Comment by guncha25** [comment]
@dustinleblanc Good point. Fix included in 8.0.5 version.
